src/vhdl/vhdl-synthesis_checks.adb

```diff
@@ -0,0 +1,165 @@
+            Real_Iir := Get_Element (Real_It);
+            if Get_Named_Entity (Real_Iir) = Gold_Iir then
+               Match := True;
+               exit;
+            end if;
+            Next (Real_It);
+         end loop;
+
+         if Match = False then
+            Warning_Msg_Sem (Warnid_Synth_Sensitivity_Lists, +Proc,
+                             "Incomplete sensitivity list! Missing signal: " &
+                             Image_Name_Id (Get_Identifier (Gold_Iir)));
+         end if;
+         Next (Gold_It);
+      end loop;
+   end Check_Incomplete_Sensitivity_List;
+
+   -- Checks if each signal from process sensitivity list are present in
+   -- "Golden sensitivity list". Throws a warning for each missing signal.
+   procedure Check_Overspecified_Sensitivity_List (Proc : Iir;
+                                                   Gold_List : Iir_List;
+                                                   Real_List : Iir_List)
+   is
+      Real_It : List_Iterator;
+      Gold_It : List_Iterator;
+
+      Gold_Iir : Iir;
+      Real_Iir : Iir;
+
+      Match : Boolean;
+   begin
+      Real_It := List_Iterate (Real_List);
+      while Is_Valid (Real_It) loop
+         Real_Iir := Get_Element (Real_It);
+         Match := False;
+         Gold_It := List_Iterate (Gold_List);
+
+         while Is_Valid (Gold_It) loop
+            Gold_Iir := Get_Element (Gold_It);
+            if Get_Named_Entity (Real_Iir) = Gold_Iir then
+               Match := True;
+               exit;
+            end if;
+            Next (Gold_It);
+         end loop;
+
+         if Match = False then
+            Warning_Msg_Sem (Warnid_Synth_Sensitivity_Lists, +Proc,
+                             "Over-specified sensitivity list! " &
+                             "Redundant signal: " &
+                             Image_Name_Id (Get_Identifier (Real_Iir)));
+         end if;
+
+         Next (Real_It);
+      end loop;
+   end Check_Overspecified_Sensitivity_List;
+
+   procedure Check_Sensitivity_Lists (Unit: Iir_Design_Unit) is
+      El : Iir;
+      Stmt : Iir;
+      Kind : Iir_Kind;
+      Gold_List : Iir_List;
+      Real_List : Iir_List;
+   begin
+      El := Get_Library_Unit (Unit);
+      if Iir_Kinds_Library_Unit (Get_Kind (El)) = Iir_Kind_Architecture_Body
+      then
+         Stmt := Get_Concurrent_Statement_Chain (El);
+         while Stmt /= Null_Iir loop
+            Kind := Get_Kind(Stmt);
+
+            -- FIXME: We should to ignore processes if we have "pragma
+            --        synthesis off" defined! Such processes will not be
+            --        synthesized, therefore shall not be checked!
+            if Kind = Iir_Kind_Sensitized_Process_Statement then
+               Gold_List := Get_Golden_Sensitivity_List (Stmt);
+               Real_List := Get_Sensitivity_List (Stmt);
+
+               -- Skip clocked processes when checking !
+               if Get_Is_Clocked_Process (Stmt) = False and
```

Comment:
I don't think the `Is_Clocked_Process` is really needed, but that's a discussion for later.

Reply:
Probably no if the detection of "golden" sensitivity list works also for clocked processes. So far I went with simple flag indicating this and just ignoring them. Later, I would like to have check of clocked processes too. This is especially important if you have reset of a flop as OR of multiple reset signals (altough it is not a good RTL practice), this typically leads to forgetting one of the reset signals in sensitivity list.